### ISSUE TITLE
Feature Request: !Sub equivalent of W1001

### ISSUE THREAD

**Issue by iamed2** [aws-cloudformation/cfn-lint#2227]
*cfn-lint version: 0.58.2*

I believe that `!Sub` parameters should be checked to see if they depend on conditional resources the same way W1001 checks this for `!Ref` (e.g. `SubCondParam.Value` should produce a warning). I suspect this is possible since E1019 checks for non-existent parameters within a `!Sub`.

```
---
AWSTemplateFormatVersion: 2010-09-09

Conditions:
  CreateContainerRepository: !Equals ["yes", "no"]

Resources:
  Repository:
    Type: AWS::ECR::Repository
    Condition: CreateContainerRepository

  SubCondParam:
    Type: AWS::SSM::Parameter
    Properties:
      Type: String
      Value: !Sub ${Repository}

  RefCondParam:
    Type: AWS::SSM::Parameter
    Properties:
      Type: String
      Value: !Ref Repository

  SubFakeParam:
    Type: AWS::SSM::Parameter
    Properties:
      Type: String
      Value: !Sub ${Fake}

  RefFakeParam:
    Type: AWS::SSM::Parameter
    Properties:
      Type: String
      Value: !Ref Fake
```

* SubCondParam.Value shows no error or warning, I believe it should show a warning
* RefCondParam.Value shows W1001
* SubFakeParam.Value shows E1019
* RefFakeParam.Value shows E1012
